doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,455 @@
+    plugin: meson
+    meson-parameters:
+      - -Ddocumentation=false
+      - -Dtests=false
+    build-packages:
+      - libxml2-dev
+      - libffi-dev
+      - libexpat1-dev
+      - pkg-config
+    override-build: |
+      snapcraftctl build
+      # We use the libwayland-egl from the vendor-drivers part.
+      # We remove this here, as otherwise snapcraft will complain that we have two
+      # different files trying to provide libwayland-egl.so
+      rm $SNAPCRAFT_PART_INSTALL/usr/local/lib/aarch64-linux-gnu/libwayland-egl*
+    source: https://gitlab.freedesktop.org/wayland/wayland.git
+    source-type: git
+    source-tag: 1.21.0
+    organize:
+      'usr/local/lib/aarch64-linux-gnu': graphics/lib/
+    prime:
+      - graphics/lib/*
+```
+
+#### Vendor driver uses a $VENDOR_dri.so DRI driver
+Some vendor drivers (internally based on Mesa) retain the separate DRI driver component. If
+a vendor driver is missing its DRI component then running a program using EGL will generate
+a `libGL` warning about failure to load `$VENDOR_dri.so`, followed by either program failure
+or possibly the program continuing with CPU rendering.
+
+Any vendor DRI driver should be provided under the top-level dri/ directory. An example
+from the `imx-gpu-viv` driver is:
+
+```yaml
+# TODO: What is the context here?
+organize:
+  imx-gpu-viv-6.4.3.p0.0-aarch64/gpu-core/usr/lib/dri/* : graphics/dri/
+```
+
+Snaps which use the `graphics-core22` interface will set the necessary environment variables
+to point the DRI loader at the top-level `dri/` path under the path exported by the
+`graphics-core22` slot.
+
+#### Vendor drivers use libglvnd
+Some drivers might use the `libglvnd` vendor-neutral GL/EGL library. In this case the vendor
+drivers will ship an `egl_vendor.d/$SOMETHING.json` file, pointing to their vendor EGL library.
+The [mesa-core22](https://github.com/canonical/mesa-core22/blob/main/snap/snapcraft.yaml)
+`snapcraft.yaml` provides an example of this sort of driver; the `egl_vendor.d`
+directory is shipped in a top-level `glvnd` path, under the path exported by the
+`graphics-core22` slot.
+
+## How to test a `graphics-core22` provider snap
+Now we have our `graphics-core22` provider snap built. However, we don't yet know if it works
+properly. Testing this snap is our next task.
+
+For starters, install the [graphics-test-tools](https://snapcraft.io/graphics-test-tools) snap:
+
+```shell
+sudo snap install graphics-test-tools --channel 22/stable
+```
+
+Next, connect the snap that you've built to `graphics-test-tools`:
+
+```shell
+sudo snap connect graphics-test-tools:graphics-core22 <your-snap>:graphics-core22
+```
+
+With that installed, we can begin testing.
+
+### Use `eglinfo` to test
+This is the most basic test that can be performed, and is a good indication of baseline GPU
+driver setup. This should list _at least_ a GBM platform with the expected vendor and 
+`OpenGL_ES` client API. An abbreviated example (on a `mesa-core22` system):
```

Comment:
@RAOF I changed this from `mesa-core20` to `mesa-core22`. Is that a problem?

Reply:
No.